### DISCUSSION TITLE
Why is there no float type？

### DISCUSSION THREAD

**Discussion by blueYangAAYY** [codingplc/plc-simulator#6] [General]
Why is there no float type？

**Comment by dagryl** [comment]
Because number type covers float.

**Reply by blueYangAAYY** [reply]
Thank you for your reply, but when I use number, I cannot change the value to a floating point number

**Comment by dagryl** [comment]
Hmm... You are right. I thought you could use floats. 

```
const setVar = (uuid: string, inputValue: string) => {
    const value = inputValue
      ? Math.max(Number(VALUE_MIN), Math.min(Number(VALUE_MAX), Number(parseInt(inputValue))))
      : 0;
    dispatch({
      type: SET_VAR_VALUE,
      payload: { uuid, value },
    });
  };
```
Probably I didn't see the need for it. What is your use case for floats?

**Reply by blueYangAAYY** [reply]
School assignments require that many PLC data be floating-point numbers

**Reply by dagryl** [reply]
I'm afraid you will have to find a different tool. PLC Simulator Online is simplified and you won't find many data types that you can in normal PLCs. :(
